libp2p/wire.nim

```diff
@@ -20,21 +20,45 @@ else:
 
 const
   TRANSPMA* = mapOr(
-    mapAnd(IP, mapEq("udp")),
-    mapAnd(IP, mapEq("tcp")),
-    mapAnd(mapEq("unix"))
+    UDP,
+    TCP,
+    UNIX,
   )
 
   RTRANSPMA* = mapOr(
-    mapAnd(IP, mapEq("tcp")),
-    mapAnd(mapEq("unix"))
+    TCP,
+    UNIX
   )
 
+proc resolveDns(ma: MultiAddress, domain: Domain = Domain.AF_UNSPEC, prefix: string = ""): MaResult[TransportAddress]
+  {.raises: [Defect, ResultError[string]]} =
```

Comment:
Probably better to map to a native exception with `toException`, similar to how it's done in `peerinfo.nim`

Reply:
Hmm, actually probably better to stick with just `Results` and not mix `tryGet`.

Reply:
yeah, things should either return `Result` or raise, but never both

Reply:
Yeah, this is coming from the `initTAddress` function. I'll wait for #582 to be merged, and rebase on it